libs/stocks/data-access-price-query/src/lib/+state/price-query.effects.ts

```diff
@@ -24,12 +24,30 @@ export class PriceQueryEffects {
       run: (action: FetchPriceQuery, state: PriceQueryPartialState) => {
         return this.httpClient
           .get(
-            `${this.env.apiURL}/beta/stock/${action.symbol}/chart/${
-              action.period
-            }?token=${this.env.apiKey}`
+            `${this.env.apiURL}/beta/stock/${action.symbol}/chart/max?token=${this.env.apiKey}`
           )
           .pipe(
-            map(resp => new PriceQueryFetched(resp as PriceQueryResponse[]))
+            map(resp => {
+              // All stocks data
+              let responseData: any = resp;
+              /**
+               * setting hours, minutes, sec and ms of Date object to 0
+               * so that we can compare dates without time
+               */
+              const timeLessDateFrom = action.dateFrom.setHours(0, 0, 0, 0);
+              const timeLessDateTo = action.dateTo.setHours(0, 0, 0, 0);
+
+              /**
+               * compare and get data between two user entered dates
+               */
+              responseData = responseData.filter(item => {
+                const timeLessStockDate = this.removeTimeFromDate(item.date, '-', '/');
```

Comment:
setting time(hours, minutes, seconds and miliseconds) of stockDate to 0 so that we can compare dates.